sdl_android/src/main/java/com/smartdevicelink/api/audio/AudioStreamManager.java

```diff
@@ -0,0 +1,192 @@
+package com.smartdevicelink.api.audio;
+
+import android.net.rtp.AudioStream;
+import android.os.Build;
+import android.support.annotation.IntDef;
+import android.support.annotation.NonNull;
+import android.support.annotation.RequiresApi;
+import android.support.annotation.StringDef;
+import android.util.Log;
+
+import com.smartdevicelink.SdlConnection.SdlSession;
+import com.smartdevicelink.protocol.enums.SessionType;
+import com.smartdevicelink.proxy.interfaces.IAudioStreamListener;
+import com.smartdevicelink.proxy.interfaces.ISdl;
+import com.smartdevicelink.proxy.interfaces.ISdlServiceListener;
+import com.smartdevicelink.proxy.rpc.enums.BitsPerSample;
+import com.smartdevicelink.proxy.rpc.enums.SamplingRate;
+
+import java.io.File;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.util.LinkedList;
+import java.util.Queue;
+
+@RequiresApi(api = Build.VERSION_CODES.JELLY_BEAN)
+public class AudioStreamManager implements ISdlServiceListener {

```

Comment:
Instead of implementing `ISdlServiceListener` which exposes the public methods to developers, you can make a private instance an add that in L71. See https://github.com/smartdevicelink/sdl_android/blob/d0097ac58cc405746f924cf57e3d596d5924c8fa/sdl_android/src/main/java/com/smartdevicelink/api/VideoStreamingManager.java#L60-L80 for reference

Reply:
This class should extend `BaseSubManager`, which provides a way for the encapsulating `SdlManager` to start each submanager and be notified when they are ready or error out.

https://github.com/smartdevicelink/sdl_android/blob/feature/issue_782_Android_Managers/sdl_android/src/main/java/com/smartdevicelink/api/BaseSubManager.java